### PR TITLE
report object name upon mangle_docstrings exception

### PR DESCRIPTION
**Pull request by jnothman**

### CODE REVIEW

**Comment by jorisvandenbossche** [comment]
Does this automatically integrate with sphinx logging? (that warnings are printed, and their verbose option). Maybe we can do `from sphinx.util import logging` instead of `import logging` (like eg in autodoc: https://github.com/sphinx-doc/sphinx/blob/master/sphinx/ext/autodoc/directive.py)

File: numpydoc/numpydoc.py [hunk 1]
```diff
@@ -124,13 +127,17 @@ def mangle_docstrings(app, what, name, obj, options, lines):
...
+        try:
+            doc = get_doc_object(obj, what, u_NL.join(lines), config=cfg,
+                                 builder=app.builder)
+            if sys.version_info[0] >= 3:
+                doc = str(doc)
+            else:
+                doc = unicode(doc)
+            lines[:] = doc.split(u_NL)
+        except:
+            logger.error('While processing numpydoc for %r', name)
```
Comment (jorisvandenbossche) [inline numpydoc/numpydoc.py]: Suggestion: add `[numpydoc] ` at the front of the message? That's also how sphinx extensions like autodoc/autosummary do it

**Comment by jnothman** [comment]
done, though that prefixing is redundant with registering the logger to the
file

**Comment by jorisvandenbossche** [comment]
> done, though that prefixing is redundant with registering the logger to the file

Hmm, yes, it seems they only do the [] for debug messages (and some info), and use the location for warnings and errors (at least it looks not fully consistent for somebody who is not familiar with it)

**Comment by rgommers** [comment]
`from sphinx.util import logging` is not present in Sphinx 1.2.3, which failed CI. I'd be fine with bumping the minimum Sphinx version.

**Comment by rgommers** [comment]
I tested this locally, LGTM. +1 for merging when CI issue is resolved.

**Comment by rgommers** [comment]
Updated the tested Sphinx versions from `latest, 1.2.3` to `latest, 1.6.5`. TravisCI is happy now, so merging.

**Pull request merged**